build/WFLY-19130_publish_Wildfly_channel_manifest.adoc

```diff
@@ -0,0 +1,169 @@
+---
+categories:
+  - wf-galleon
+  - maven
+---
+= [Community] Publish Wildfly channel manifest
+:author:            Bartosz Spyrko-Smietanko
+:email:             bspyrkos@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+https://github.com/wildfly-extras/wildfly-channel[Wildfly-channels] added ability to list create channels defining component versions used to provision Wildfly.
+This ability is being used in component testing and by provisioning projects like Galleon plugins and Prospero.
+
+This proposal describes how the Wildfly manifests should be published at build time.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-19130[WFLY-19130]
+
+=== Related Issues
+
+
+
+=== Stability Level
+// Choose the planned stability level for the proposed functionality
+* [ ] Experimental
+
+* [ ] Preview
+
+* [X] Community
+
+* [ ] default
+
+=== Dev Contacts
+
+* mailto:{bspyrkos@redhat.com}[Bartosz Spyrko-Smietanko]
+
+=== QE Contacts
+
+* N/A
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+* [X] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+* https://github.com/wildfly/wildfly[WildFly]
+* https://github.com/wildfly-extras/prospero[Prospero]
+
+=== Other Interested Projects
+N/A
+
+=== Relevant Installation Types
+// Remove the x next to the relevant field if the feature in question is not relevant
+// to that kind of WildFly installation
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [x] Managed domain
+
+* [x] OpenShift s2i
+
+* [x] Bootable jar
+
+== Requirements
+
+WildFly project currently creates three feature packs: `wildfly-ee-galleon-pack`, `wildfly-galleon-pack` and `preview-galleon-pack`.
+Those feature packs can require different versions of the same component being provisioned. To provide this flexibility each
+feature packs has to have a separate channel associated with it.
+
+A channel consists of a stream of manifests with the current manifest being determined by its Maven version. When a new version
+of WildFly is published a new version of the manifest will be added to the channel.
+The backwards compatibility of a channel is achieved through a naming convention where the supported version range is added to the 
+artifactId of the manifest (e.g. `org.wildfly:wildfly-32.0`).
+
+For this proposal we will publish the latest manifest with no backwards compatibility guarantee, meaning each channel will always
+contain components of the latest released version of WildFly.
+
+Following three channels will be created as part of this proposal:
+
+[cols=3]
+|===
+| Manifest GA
+| Artifacts
+| Depends on
+
+|`org.wildfly.channels:wildfly-ee`
+|wildfly-ee-galleon-pack and its dependencies
+| -
+
+|`org.wildfly.channels:wildfly`
```

Comment:
I'm wondering if a better name would be something like `org.wildfly.channels:wildfly-full`

Reply:
That would be a bummer.

If you to provision what almost everyone regards as WildFly you use the 'wildfly' feature pack. Adding "full" is confusing.

The channels are a bit different from the feature packs though; using the 'wildfly' FP means you unavoidably get 'wildfly-ee' with no need for the user to even be aware that's happened.

Reply:
It's mostly for consistency - the other two channels have a name of `wildfly-*`, so it feels like channel name `wildfly` misses something :)
On the other hand, like you said, `wildfly` is consistent with naming of the feature packs.